SwiftyStoreKit/InAppReceipt.swift

```diff
@@ -235,5 +241,69 @@ internal class InAppReceipt {
             }
             task.resume()
     }
-    
+
+    /**
+     *  Verify the purchase of a product in a receipt
+     *  - Parameter productId: the product id of the purchase to verify
+     *  - Parameter inReceipt: the receipt to test in
+     *  - Parameter validUntil: the expires date of the subscription must be valid until this date. If nil, no verification
+     */
+    class func verifyPurchase(
+      productId productId: String,
+      inReceipt receipt: ReceiptInfo,
+      validUntil: NSDate? = nil
+    ) -> SwiftyStoreKit.VerifyPurchaseResult {
+      
+          // Get all receipts
+          guard let allReceipts = receipt["receipt"]?["in_app"] as? [ReceiptInfo] else {
+              return .NotPurchased
+          }
+      
+          // Filter receipts with matching product id
+          let receiptsMatchingProductId = allReceipts
+          .filter { (receipt) -> Bool in
+                let product_id = receipt["product_id"] as? String
+                return product_id == productId
+          }
+      
+          // Verify that at least one receipt has the right product id
+          guard receiptsMatchingProductId.count >= 1 else {
+              return .NotPurchased
+          }
+      
+          guard let validUntil = validUntil else {
+              // Do not need to verify the expiration date
+              return .Purchased(expiresDate: nil)
+          }
+      
+          // Return the expires dates sorted desc
+          let expiresDates = receiptsMatchingProductId
+          .map { (receipt) -> NSDate in
+            // TODO: should use 'expires_date' or 'expiration_date' instead? See:
+            // https://developer.apple.com/library/mac/releasenotes/General/ValidateAppStoreReceipt/Chapters/ReceiptFields.html
+              let expires_date = receipt["expires_date_ms"] as? NSString
+              let expires_date_double = (expires_date?.doubleValue ?? 0.0) / 1000
+              return NSDate(timeIntervalSince1970: expires_date_double)
```

Comment:
The expires_date related fiels are only present for Automatically Renewable Subscription.
That's why the parameter `validUntil`is optional:
- If you want to test a consumable, non-consumable or non-renewable subscription, you should let the `validUntil` parameter to `nil`. In those case, the function return `.Purchased(expiresDate: nil)`.
- If you indicate a `validUntil`parameter to test a Automatically Renewable Subscription, the function will continue after the line 278 and test the `expires_date` with the `validUntil`.

I think the user's API should be split in 2 to avoid confusion:
- One only for Automatically Renewable Subscription with a required validUntil date.
- One for the others type of product with no validUntil parameter.
Those 2 API will use the same function in the background.


That's true that `expires_date_ms` is not in the docs, but it is present in the receipt json, as well as 'purchase_date_ms', 'original_purchase_date_ms', 'expires_date_pst', `original_purchase_date_pst` and `purchase_date_pst'. I think they didn't put them in the doc because they are just helper to parse the plain string date easier.
We can parse `expires_date` with a `NSDateFormatter` if you think it's better to use only fields present in the docs.